### PR TITLE
[Android] Add background video playback

### PR DESCRIPTION
**Pull request by simonhong**
Issue: https://github.com/brave/brave-browser/issues/6293
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
@AlexeyBarabash Can you help why this PR doesn't work?
I referenced your PR(https://github.com/brave/browser-android-tabs/pull/464) and moved script injection logic to c++ layer by using tab helper. But, video is paused when brave goes to background.

**Comment by AlexeyBarabash** [comment]
@simonhong 
since PR https://github.com/brave/browser-android-tabs/pull/464 YT changed logic of how the page does pause when it goes to background. We had to change the script several times.

Finally the injected script should look as
https://github.com/brave/browser-android-tabs/blob/base-76.0.3809.132/chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java#L50

Hope this will help.

**Comment by simonhong** [comment]
@AlexeyBarabash Oh,, script was changed. thanks a lot! Will check :)

File: browser/android/preferences/background_video_playback_tab_helper.cc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline browser/android/preferences/background_video_playback_tab_helper.cc:1]: why is this file inside preferences?

File: browser/android/preferences/background_video_playback_tab_helper.cc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/navigation_controller.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/renderer/render_frame.h"
+#include "content/public/browser/navigation_handle.h"
+#include "url/gurl.h"
+
+namespace {
+const char k_youtube_background_playback_script[] =
```
Comment (bridiver) [inline browser/android/preferences/background_video_playback_tab_helper.cc:23]: what is this doing?

**Pull request merged**